Sources/SwiftSyntax/SyntaxParser.swift

```diff
@@ -66,7 +69,9 @@ public enum SyntaxParser {
   /// - Throws: `ParserError`
   public static func parse(
     source: String,
-    parseTransition: IncrementalParseTransition? = nil
+    file: String = "",
+    parseTransition: IncrementalParseTransition? = nil,
```

Comment:
Could you add documentation comments for the new parameters ?

Also since `file` is only needed for the emission of diagnostics I think it would more clear if it is named `filenameForDiagnostics`, otherwise people may get confused and think there must be some file present in order to parse some source text. 
And a bit better for the order of parameters to be
* `parseTransition`
* `filenameForDiagnostics`
* `diagnosticConsumer` or `diagnosticEngine`